src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -111,12 +111,15 @@ class InstanceOpFactoryImpl(
   }
 
   private[this] def inferForResidents(app: AppDefinition, request: InstanceOpFactory.Request): OfferMatchResult = {
-    val InstanceOpFactory.Request(runSpec, offer, instances, additionalLaunches) = request
+    val InstanceOpFactory.Request(runSpec, offer, instances, additionalLaunches, _) = request
 
     // TODO(jdef) pods should be supported some day
 
     val needToLaunch = additionalLaunches > 0 && request.hasWaitingReservations
     val needToReserve = request.numberOfWaitingReservations < additionalLaunches
+    val instances0 = request.instanceMap.mapValues(i => i.instanceId + " " + i.state.condition).mkString(", ")
+
+    log.debug(s"Need to launch $needToLaunch, need to reserve $needToReserve, additionalLaunches $additionalLaunches, instances [$instances0]")
```

Comment:
Since `instances0` is used only for debug logging, maybe wrap them in `log.isDebugEnabled` if-statement?